mono/utils/win64.asm

```diff
@@ -38,6 +38,36 @@ __mono_current_ip:
 
 mono_context_get_current endP
 
+copy_stack_data_internal_win32_wrapper PROC PUBLIC
+;rcx MonoThreadInfo *
+;rdx MonoStackData *
+;r8 MonoBuiltinUnwindInfo *
+;r9 CopyStackDataFunc
+
+	movaps xmmword ptr [r8 + 00h], xmm6
+	movaps xmmword ptr [r8 + 10h], xmm7
+	movaps xmmword ptr [r8 + 20h], xmm8
+	movaps xmmword ptr [r8 + 30h], xmm9
+	movaps xmmword ptr [r8 + 40h], xmm10
+	movaps xmmword ptr [r8 + 50h], xmm11
+	movaps xmmword ptr [r8 + 60h], xmm12
+	movaps xmmword ptr [r8 + 70h], xmm13
+	movaps xmmword ptr [r8 + 80h], xmm14
+	movaps xmmword ptr [r8 + 90h], xmm15
+
+	mov qword ptr [r8 + 0A0h], rbx
+	mov qword ptr [r8 + 0A8h], rsi
+	mov qword ptr [r8 + 0B0h], rdi
+	mov qword ptr [r8 + 0B8h], r12
+	mov qword ptr [r8 + 0C0h], r13
+	mov qword ptr [r8 + 0C8h], r14
+	mov qword ptr [r8 + 0D0h], r15
+	mov qword ptr [r8 + 0D8h], rbp
+
+	jmp r9
```

Comment:
maybe a comment "; tailcall" or maybe this is the most obvious line of assembly ever :)

Reply:
Oh, and that you are passing through the parameters. All quite subtle and simple at the same time.